teps/0122-complete-build-instructions-and-parameters.md

```diff
@@ -0,0 +1,241 @@
+
+For example the following provenance snippet was generated by Tekton Chains ([v0.8.0](https://github.com/tektoncd/chains/releases/tag/v0.8.0)) from [this chains example Task](https://github.com/tektoncd/chains/blob/main/examples/kaniko/kaniko.yaml) - it contains step information (including the shas of the images run) but is missing other information such as the workspaces and the volumes that backed them:
+
+```yaml
+invocation:
+  configSource: {}
+  parameters:
+    BUILDER_IMAGE: gcr.io/kaniko-project/executor:v1.5.1@sha256:c6166717f7fe0b7da44908c986137ecfeab21f31ec3992f6e128fff8a94be8a5
+    CONTEXT: ./
+    DOCKERFILE: ./Dockerfile
+    EXTRA_ARGS: '[]'
+    IMAGE: '{string us.gcr.io/christiewilson-catfactory/kaniko-chains []}'
+buildConfig:
+  steps:
+    - entryPoint: |
+        set -e
+        echo "FROM alpine@sha256:69e70a79f2d41ab5d637de98c1e0b055206ba40a8145e7bddb55ccc04e13cf8f" | tee $(params.DOCKERFILE)
+      arguments: null
+      environment:
+        container: add-dockerfile
+        image: docker-pullable://bash@sha256:fc742d0c3d9d8f5fb2681062398c04b710cd08c46dac1a8f0a5515687018acb9
+      annotations: null
+    - entryPoint: ""
+      arguments:
+        - $(params.EXTRA_ARGS)
+        - --dockerfile=$(params.DOCKERFILE)
+        - --context=$(workspaces.source.path)/$(params.CONTEXT)
+        - --destination=$(params.IMAGE)
+        - --digest-file=$(results.IMAGE_DIGEST.path)
+      environment:
+        container: build-and-push
+        image: docker-pullable://gcr.io/kaniko-project/executor@sha256:68bb272f681f691254acfbdcef00962f22efe2f0c1e287e6a837b0abe07fb94b
+      annotations: null
+    - entryPoint: |
+        set -e
+        echo $(params.IMAGE) | tee $(results.IMAGE_URL.path)
+      arguments: null
+      environment:
+        container: write-url
+        image: docker-pullable://bash@sha256:fc742d0c3d9d8f5fb2681062398c04b710cd08c46dac1a8f0a5515687018acb9
+      annotations: null
+```
+
+## Requirements
+- For completeness of build instructions, the provenance should contain all of the Task definition and runtime information.
+- The shas of the images run should continue to be included regardless (as this is part of the dependencies / build instructions)
+- It must be possible for users to construct policies based on the provenance that would allow them to determine if the build is ok - for example if they have policies around what pipeline tasks are acceptable to use, what parameters are acceptable, whether or not sidecars are allowed etc.
+  - According to [SLSA v0.2](https://slsa.dev/provenance/v0.2), if **invocation.configSource** is not available then **buildConfig** can be used to verify information about the build. If the policy requires validating the underlying task/pipeline spec that was not fetched from version control then the information from the spec required for validation must be embedded in the provenance’s buildConfig. 
```

Comment:
If I understand this correctly, when the resource definition is not coming from a versioned source, we use `buildConfig` to store it directly in the provenance, correct?
Would it make sense later on in the spec to specify the list of sources that are considered ok for `configSource`?

Reply:
>If I understand this correctly, when the resource definition is not coming from a versioned source, we use buildConfig to store it directly in the provenance, correct?
>
Yes, that is correct.
> Would it make sense later on in the spec to specify the list of sources that are considered ok for configSource?
> 
My understanding is that SLSA needs the references need to be immutable. We support three: GitHub, OCI bundles and hub.